core/time/timezone/tz_unix.odin

```diff
@@ -0,0 +1,87 @@
+#+build darwin, linux, freebsd, openbsd, netbsd
+#+private
+package timezone
+
+import "core:os"
+import "core:strings"
+import "core:path/filepath"
+import "core:time/datetime"
+
+local_tz_name :: proc(allocator := context.allocator) -> (name: string, success: bool) {
+	local_str, ok := os.lookup_env("TZ", allocator)
+	if !ok {
+		orig_localtime_path := "/etc/localtime"
+		path, err := os.absolute_path_from_relative(orig_localtime_path, allocator)
+		if err != nil {
+			// If we can't find /etc/localtime, fallback to UTC
+			if err == .ENOENT {
+				str, err2 := strings.clone("UTC", allocator)
+				if err2 != nil { return }
+				return str, true
+			}
+
+			return
+		}
+		defer delete(path, allocator)
+
+		// FreeBSD makes me sad.
```

Comment:
This is an unfortunate hackaround, because FreeBSD, unlike other distros, copies, rather than softlinks their timezone file, and then puts the original name of the timezone in another file at /var/db/zoneinfo.